@here/harp-mapview/lib/poi/PoiRenderer.ts

```diff
@@ -541,6 +546,82 @@ export class PoiRenderer {
         this.m_poiBatchRegistry.update();
     }
 
+    /**
+     * @internal
+     *
+     * Adds a layer to the PoiRenderer
+     * @param layerId
+     */
+    addLayer(layerId: number): PoiLayer {
+        let result = this.getLayer(layerId);
+        if (result === undefined) {
+            result = {
+                id: layerId,
+                scene: new THREE.Scene()
+            };
+
+            this.m_layers.push(result);
+            this.m_layers.sort((a: PoiLayer, b: PoiLayer) => {
+                return a.id - b.id;
+            });
+        }
+        return result;
+    }
+
+    /**
+     * Retrieves a specific `Poi` rendering layer.
+     *
+     * @param layerId - Desired layer identifier.
+     *
+     * @returns Selected {@link PoiLayer}
+     */
+    private getLayer(layerId: number): PoiLayer | undefined {
+        return this.m_layers.find(layer => layer.id === layerId);
+    }
+
+    /**
+     * Renders the content of this `PoiRenderer`.
+     *
+     * @param camera - Orthographic camera.
+     * @param betweenLayersCallback - Optional callback between layers.
+     * @param target - Optional render target.
+     * @param clear - Optional render target clear operation.
+     */
+    render(
+        camera: THREE.OrthographicCamera,
+        betweenLayersCallback?: (
+            lastRendererdLayerId?: number,
+            nextRenderedLayerId?: number
+        ) => void,
+        target?: THREE.WebGLRenderTarget,
+        clear?: boolean
+    ) {
+        let oldTarget: THREE.RenderTarget | null = null;
+        if (target !== undefined) {
+            oldTarget = this.m_renderer.getRenderTarget();
+            this.m_renderer.setRenderTarget(target);
+        }
+        if (clear === true) {
+            this.m_renderer.clear(true);
+        }
+        if (this.m_layers.length <= 0) {
+            betweenLayersCallback?.();
+        } else {
+            for (let i = 0; i < this.m_layers.length; i++) {
+                const layer = this.m_layers[i];
+                if (i === 0) {
+                    betweenLayersCallback?.(undefined, layer.id);
+                }
+                this.update();
```

Comment:
Why do we need to do this here multiple times instead of doing it once at the end of placeText like before?
Isn't there some side effect of updating the BoxBuffer's multiple times?

Reply:
done